src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,443 @@
+    }
+
+
+    /**
+     * @return string
+     */
+    public function getValidationUrl()
+    {
+        return $this->getParameter('validationUrl');
+    }
+
+    /**
+     * Set the validation URL for the request
+     *
+     * @param string $url
+     * @return static
+     */
+    public function setValidationUrl($url)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('validationUrl', $url);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getMerchantIdentifier()
+    {
+        return $this->getParameter('merchantIdentifier');
+    }
+
+    /**
+     * Set the merchant identifier for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setMerchantIdentifier($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('merchantIdentifier', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getDisplayName()
+    {
+        return $this->getParameter('displayName');
+    }
+
+    /**
+     * Sets the display name for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setDisplayName($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('displayName', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getApplicationUrl()
+    {
+        /**
+         * @var null|string
+         */
+        return $this->getParameter('applicationUrl');
+    }
+
+    /**
+     * Sets the application url or the "initiativeContext" for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setApplicationUrl($value)
+    {
+        /**
+         * @var static
+         */
+        return $this->setParameter('applicationUrl', $value);
+    }
+
+    /**
+     * Get HTTP Method.
+     *
+     * This is nearly always POST but can be over-ridden in sub classes.
+     *
+     * @return string
+     */
+    public function getHttpMethod()
+    {
+        return 'POST';
+    }
+
+    /**
+     * @return array
+     */
+    public function getData()
+    {
+        // This ensures that the validation URL set. If it isn't, the request call won't be sent.
+        $this->validate('validationUrl');
+        $data = array();
+
+        // Required request parameters for Apple Session object.
+        $data['merchantIdentifier'] = $this->getMerchantIdentifier();
+        $data['displayName'] = $this->getDisplayName();
+        // Default parameter for Apple Pay on the Web.
+        $data['initiative'] = 'web';
+        $data['initiativeContext'] = $this->getApplicationUrl();
+
+        return $data;
+    }
+
+    /**
+     * @param       $data
+     * @param array $headers
+     *
+     * @return static
+     */
+    protected function createClientRequest($data, array $headers = null)
```

Comment:
Please add docblock for this function, and it would be better if you could write down what kind of role this is playing in the payment flow